### ISSUE TITLE
Fix pyflakes for web

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7000]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#7000|
|Type|enhancement|
|Created|2014-02-27 13:40:01Z|
|Branch|https://github.com/twisted/twisted/tree/twisted-web-pyflakes-7000-3|

Attachments:

* [7000-1.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/b75/b7525fa5653c5b45f75d490825626e0880b394f4/faa850c8ded89036660df6dd6670b2035efc75c0.diff) (9054 bytes) - added by adiroiban on 2014-02-27 13:50:28Z - 
* [7000-2.dif](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/b75/b7525fa5653c5b45f75d490825626e0880b394f4/a8ba0861c87c5c3d8307151fa4dfb13d83018599.dif) (15645 bytes) - added by adiroiban on 2014-04-14 08:46:36Z - 
* [7000-3.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/b75/b7525fa5653c5b45f75d490825626e0880b394f4/7bb0ae1da0cd54cf419061c34ea6bd8409b223e3.diff) (15649 bytes) - added by adiroiban on 2014-11-10 17:01:28Z - 
* [7000-4.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/b75/b7525fa5653c5b45f75d490825626e0880b394f4/f99d13017d07044edf9a4315310ee6b78e708454.diff) (15649 bytes) - added by adiroiban on 2014-11-10 17:18:51Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7000 7000
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_twisted_web_pyflakes_7000_3 branches/twisted-web-pyflakes-7000-3
branch_author__hawkowl__adiroiban__habnabit hawkowl, adiroiban, habnabit
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1393508401000000 1393508401000000
changetime__1423931536529723 1423931536529723
version__None None
owner__adiroiban adiroiban
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

NodeList was not used in any part and was removed.

All other should be just removal of unused local variables or unused imports.

**Comment by twisted-trac** [comment]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)<a name="note_3"></a>|@habnabit commented|
|-|-|

(In [42190]) Branching to 'twisted-web-pyflakes-7000'

**Comment by twisted-trac** [comment]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)<a name="note_4"></a>|@habnabit commented|
|-|-|

(In [42191]) Apply patch from adiroiban. refs [#7000](https://github.com/twisted/twisted/issues/7000)

**Comment by twisted-trac** [comment]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)<a name="note_5"></a>|@habnabit set owner to @adiroiban|
|-|-|

Thanks for the contribution!

Unfortuantely, [this totally breaks the tests](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/twisted-web-pyflakes-7000). Some of the names weren't used within the modules but were used indirectly.

 1. `twisted.web.test.test_xmlrpc` should probably be using [requireModule](https://github.com/twisted/twisted/commit/97b5d6b377b9d308d4e7d7dc598b7fa745ac1cc9) instead of a `try-except`.
 2. `twisted.web.test.test_xml` should just call `hash` directly instead of making a temporary dictionary.
 3. `twisted.web.test.test_http` doesn't need anything after `# Silence the linter.` in the `try` block; I think you're running an old pyflakes.
 4. `twisted.web.tap`: I raised this in dev irc, but there should be a standard idiom for checking for SSL support. Importing from `twisted.internet.ssl` probably isn't necessary; you could use `requireModule` on `OpenSSL.SSL` instead.
 5. Finally, your patch needs a [news file](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles).

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban removed owner|
|-|-|


>  1. `twisted.web.test.test_xmlrpc` should probably be using
> [https://github.com/twisted/twisted/commit/97b5d6b377b9d308d4e7d7dc598b7fa745ac1cc9
> requireModule] instead of a `try-except`.

I kept try/except and used namedAny.
requireModule was not designed to be used as direct 'skip' attribute...it
does the reverse.

>  2. `twisted.web.test.test_xml` should just call `hash` directly instead
> of making a temporary dictionary.

I have updated the test and tried to add a docstring. Not sure what is the
exact purpose of that test.

>  3. `twisted.web.test.test_http` doesn't need anything after `# Silence
> the linter.` in the `try` block; I think you're running an old pyflakes.

True. I was using an old pyflakes. Updated.

>  4. `twisted.web.tap`: I raised this in dev irc, but there should be a
> standard idiom for checking for SSL support. Importing from
> `twisted.internet.ssl` probably isn't necessary; you could use
> `requireModule` on `OpenSSL.SSL` instead.

Thanks. I went for requireModule('OpenSSL.SSL')

>  5. Finally, your patch needs a
> [news file](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles).

Done.


-----

In client.py I removed the double import of failure and kept only the
direct import to Failure... in this way 'failure' variable can be used
for local work.

In soap.py ns variable was not used.

In twisted/web/domhelpers.py I had no better idea of how to fix the linter.
One option is to define ```__all___``` variable but I don't like that
variable as is easy to get it out of sync and forget adding new things
to it.

Run all web tests and pyflakes on local computer without errors.

I have attached a new patch based on latest trunk as somehow I got new errors... but maybe they are just because pyflakes upgrade.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl commented|
|-|-|

(In [43344]) Branching to twisted-web-pyflakes-7000-2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_9"></a>|@hawkowl commented|
|-|-|

(In [43345]) apply patch from adiroiban. refs [#7000](https://github.com/twisted/twisted/issues/7000)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl set owner to @adiroiban|
|-|-|

Hi Adi, thanks for this! Line numbers are taken from https://github.com/twisted/twisted/compare/trunk...twisted-web-pyflakes-7000-2.

- There's not a newsfile in the patch.
- In twisted/web/tap.py line 174, it will only raise UsageError if the import fails. However, because requireModule eats the exception, as it is designed to, do, this UsageError cannot ever be reached. Could you change this to an if statement instead maybe?
- I think the same is true of  twisted/web/test/test_xmlrpc.py line 25 -- I don't think that UsageError will ever be reached again, since NamedAny doesn't raise ImportError.

Otherwise, everything looks good. Could you fix up these minor things, then tag it for review and assign it to me, and I'll merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

> In twisted/web/tap.py line 174, it will only raise UsageError if the import fails. However, because requireModule eats the exception, as it is designed to, do, this UsageError cannot ever be reached. Could you change this to an if statement instead maybe?

Sounds like there is also a missing test?

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_12"></a>|@adiroiban commented|
|-|-|

Hi and thanks for the review.

* The topfile is there in my patch... Trac has a problem and it don't show empty files in diff

* Yep. On line 174 I should have used namedModule.

* Yep on line 23 I should have used namedModule.

---------

I have created [#7709](https://github.com/twisted/twisted/issues/7709) to write a test for postOptions

------

I have attached a new diff based on latest trunk.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)|@adiroiban removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

> I have created [[#7709](https://github.com/twisted/twisted/issues/7709)](https://github.com/twisted/twisted/issues/7709) to write a test for postOptions

Test **driven** development, please. :(

I do not think it is okay to punt this to a later ticket.  Please resolve #7709 first if you want to resolve it separately from this work.

Thank you.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_15"></a>|@adiroiban commented|
|-|-|

But I was not the one who wrote the code for 'postOptions' ... I just had the bad luck to break it :)

I have attached a new diff which does not touch the tap.py code... leaving it to be fixed in [#7709](https://github.com/twisted/twisted/issues/7709)

I think that this ticket should be about cleaning code of pyflakes errors and not writing missing tests.

Cheers!

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_16"></a>|@wsanchez set owner to @adiroiban|
|-|-|

`7004-4.diff` looks good, tests pass.

adiroiban: You're good to go.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_17"></a>|@adiroiban commented|
|-|-|

(In [43846]) Branching to twisted-web-pyflakes-7000-3.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_18"></a>|@adiroiban commented|
|-|-|

I don't know where  to look for build results for 7004-4.diff ... I have created a new branch and will send it to buildbot... will merge if all is ok thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_19"></a>|@adiroiban commented|
|-|-|

Also, I need to fix [#7709](https://github.com/twisted/twisted/issues/7709) first as asked by exarkun

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_20"></a>|@adiroiban removed owner|
|-|-|

I have fixed it here, as I think is the right thing to do.

I was close to miss exarkun's comment.

maybe reviewers should update the description and state if they approve or not latet changes. Should be much easier to see status of a ticket, rather than check each comment and see who is reviewer and if he/she has approved the changes... still this is not an excuse for not reading all comments

to test the changes use:

```
./bin/trial --without-module=OpenSSL twisted/web/test/test_tap.py
./bin/trial twisted/web/test/test_tap.py
```

please take a look at latest changes.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_21"></a>|@wsanchez set owner to @adiroiban|
|-|-|

Nice.  Looks great, thanks.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_22"></a>|@adiroiban set status to `closed`|
|-|-|

(In [43929]) Merging twisted-web-pyflakes-7000-3: Clean pyflakes for twisted.web.

Author: adiroiban
Reviewer: wsanchez, exarkun, hawkowl, habnabit
Fixes: [#7000](https://github.com/twisted/twisted/issues/7000)
